open3d_slam/src/Submap.cpp

```diff
@@ -129,8 +129,8 @@ void Submap::carve(const PointCloud &scan, const Eigen::Vector3d &sensorPosition
 	if (cloud->empty() || !(nScansInsertedDenseMap_ % param.carveSpaceEveryNscans_ == 1)) {
 			return;
 		}
-//	auto croppedScan = removeDuplicatePointsWithinSameVoxels(scan, Eigen::Vector3d::Constant(params_.denseMapBuilder_.mapVoxelSize_));
-	std::vector<Eigen::Vector3i> keysToRemove = getKeysOfCarvedPoints(scan, *cloud, sensorPosition, param);
+	auto croppedScan = removeDuplicatePointsWithinSameVoxels(scan, Eigen::Vector3d::Constant(params_.denseMapBuilder_.mapVoxelSize_));
```

Comment:
personally I think it is good to indicate in the name that it is a ptr, in particular as you used "auto".
--> use croppedScanPtr

what do you think?

Reply:
Changed.